WeatherApp/ViewControllers/WeatherListViewController.swift

```diff
@@ -48,29 +55,87 @@ class WeatherListViewController: UIViewController {
         view.setupGradientBackground()
     }
     
-    private func setupTableView() {
-        tableView.rx.setDelegate(self).disposed(by: disposeBag)
-        tableView.register(WeatherTableViewCell.self, forCellReuseIdentifier: WeatherTableViewCell.identifier)
+    override func viewWillAppear(_ animated: Bool) {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .search, target: self, action: #selector(showSearchBar))
     }
     
-    private func bindTableView() {
-        weatherViewModel
-            .currentWeatherList
-            .bind(to: tableView.rx.items(cellIdentifier: WeatherTableViewCell.identifier, cellType: WeatherTableViewCell.self)) { [weak self] (row, currentWeather, cell) in
+    private func createTimer() {
+        timerDisposeBag = DisposeBag()
+        
+        Observable<Int>
+            .timer(.seconds(0), period: .seconds(600), scheduler: MainScheduler.instance)
+            .subscribe(onNext: { [weak self] _ in
                 guard let self = self else { return }
                 
-                self.endRefreshing()
-                cell.setup(currentWeather)
-        }
-        .disposed(by: disposeBag)
+                self.weatherViewModel.refreshData.onNext(())
+            })
+            .disposed(by: timerDisposeBag)
+    }
+    
+    private func createDataSource() {
+        dataSource = RxTableViewSectionedReloadDataSource<SectionOfCurrentWeather>(
+            configureCell: { _, tableView, indexPath, item in
+                let cell = tableView.dequeueReusableCell(withIdentifier: WeatherTableViewCell.identifier, for: indexPath) as! WeatherTableViewCell
+                cell.setup(item)
+                self.refreshControl.endRefreshing()
+                return cell
+        }, canEditRowAtIndexPath: {_,_ in
+            return true
+        })
+    }
+    
+    private func bindTableView() {
+        weatherViewModel.currentWeatherList
+            .bind(to: tableView.rx.items(dataSource: dataSource))
+            .disposed(by: disposeBag)
         
-        tableView.rx.modelSelected(CurrentWeather.self)
+        tableView.rx
+            .modelSelected(CurrentWeather.self)
             .subscribe(
                 onNext: { [weak self] (currentWeather) in
                     guard let self = self else { return }
                     
                     self.weatherViewModel.modelSelected.onNext(currentWeather)
             }).disposed(by: disposeBag)
+        
+    }
+    
+    private func bindSearchBar() {
+        searchBar.rx.cancelButtonClicked.subscribe(onNext: { [weak self] (_)  in
```

Comment:
searchBar
    .rx
    .cancelButtonClicked
    .subscribe(...)

is more readable in my opinion.